### PR TITLE
 codegen: Improve the assertion message of failing layout tests.

### PR DESCRIPTION
**Pull request by emilio**
r? @fitzgen 

cc @flier

### CODE REVIEW

**Comment by emilio** [comment]
Actually, @flier, wanna review this? :)

File: src/codegen/mod.rs [hunk 1]
```diff
@@ -2745,7 +2745,7 @@ mod utils {
     }
 
     pub fn fnsig_return_ty(ctx: &BindgenContext,
-                           sig: &super::FunctionSig)
+                           sig: &FunctionSig)
```
Comment (fitzgen) [inline src/codegen/mod.rs]: If we are going to support older rustc, we should add the oldest supported version to our travis ci matrix.

Comment (emilio) [inline src/codegen/mod.rs]: This is only temporary until Gecko updates is rustc version anyway, should be <1week

File: src/codegen/mod.rs [hunk 2]
```diff
@@ -866,8 +866,10 @@ impl CodeGenerator for CompInfo {
                 let item = quote_item!(ctx.ext_cx(),
                                        #[test]
                                        fn $fn_name() {
-                                           assert_eq!($size_of_expr, $size);
-                                           assert_eq!($align_of_expr, $align);
+                                           assert_eq!($size_of_expr, $size,
+                                                      concat!("Size of template specialization: ", stringify!($ident)));
+                                           assert_eq!($align_of_expr, $align,
+                                                      concat!("Alignment of template specialization: ", stringify!($ident)));
                                        })
```
Comment (flier) [inline src/codegen/mod.rs]: The patch is good to me, something else observed. 

Do we need check the field offset for the template specialization?

It seems be different to VTable case, should have potential layout issue. Anyway, I haven't user cases in wild

**Pull request merged**